### PR TITLE
Enforce limit of 1024 fields

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

**Comment by emmanuelkeller** [comment]
That sounds good !

Elastic has a default limit set to 1000, but configurable.
https://www.elastic.co/guide/en/elasticsearch/reference/master/mapping-settings-limit.html

Shall we do it a configurable limit ? It would be nice to not block a use case because they would have 1026 fields ..

@MeirShpilraien WDYT ?

File: src/spec.c [hunk 1]
```diff
@@ -418,23 +418,28 @@ static int IndexSpec_AddFieldsInternal(IndexSpec *sp, ArgsCursor *ac, QueryError
   FieldSpec *fs = NULL;
 
   while (!AC_IsAtEnd(ac)) {
-    size_t nfieldName = 0;
+    size_t nfieldName;
     const char *fieldName = AC_GetStringNC(ac, &nfieldName);
     if (IndexSpec_GetField(sp, fieldName, nfieldName)) {
-      QueryError_SetError(status, QUERY_EINVAL, "Duplicate field in schema");
+      QueryError_SetErrorFmt(status, QUERY_EINVAL, "Duplicate field in schema - %s", fieldName);
```
Comment (emmanuelkeller) [inline src/spec.c:424]: Regarding security, is there any risk of overflow when passing the in QueryError_SetErrorFmt  fieldName ?

Comment (ashtul) [reply src/spec.c:424]: @emmanuelkeller 
No. The function calls `rm_vasprintf` and creates another string that is later freed.

**Comment by ashtul** [comment]
@emmanuelkeller @MeirShpilraien 
Any index creates an `RSSortingTable` which has a static array of `sortField` structs with a length of `RS_SORTABLES_MAX`. with the new value of `1024` it will take additional ~12kb per index.
Should we change it to be dynamic? It will save up to 4kb per index

**Review by rafie — APPROVED** [review]

**Pull request merged**